tests/toolchain/gcc5_C_compilation.pm

```diff
@@ -0,0 +1,42 @@
+use base "opensusebasetest";
+use strict;
+use testapi;
+
+sub run() {
+    my $self = shift;
+
+    type_string "wget https://github.com/linux-test-project/ltp/releases/download/20150420/ltp-full-20150420.tar.bz2 && echo 'OK'\n";
+    type_string "tar jxf ltp-full-20150420.tar.bz2\n";
+    sleep 2;
+    type_string "export CC=/usr/bin/gcc-5\n";
+    type_string "export CXX=/usr/bin/g++-5\n";
+    type_string "cd ltp-full-20150420\n";
+    sleep 2;
+    type_string "setterm -blank 0\n";   # disable screensaver
+    type_string "./configure --with-open-posix-testsuite && echo 'configure PASSED-$?'|tee /dev/$serialdev\n";
```

Comment:
Ok, now when you check the logs what you actually write you get:
`<<< type_string(string='./configure --with-open-posix-testsuite && echo \'configure PASSED-0\'|tee /dev/ttyS0
', max_interval=250)`
No $? there, since perl replaced that before passing to type_string ;)
You need to escape \$?.
Btw. why you just don't use assert_script_run? It will even die for you :)